### PR TITLE
[rapids] Update to 0.3 version of spark-rapids and 0.17 libcudf

### PR DESCRIPTION
**Pull request by sameerz**
Signed-off-by: Sameer Raheja <sraheja@nvidia.com>

cc: @mengdong

### CODE REVIEW

**Review by functicons — APPROVED** [review]

File: rapids/rapids.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ readonly RUN_WORKER_ON_MASTER=$(get_metadata_attribute 'dask-cuda-worker-on-mast
 readonly DEFAULT_CUDA_VERSION="10.2"
 readonly CUDA_VERSION=$(get_metadata_attribute 'cuda-version' ${DEFAULT_CUDA_VERSION})
 readonly CUDF_VERSION=$(get_metadata_attribute 'cudf-version' ${DEFAULT_CUDF_VERSION})
-readonly RAPIDS_VERSION=$(get_metadata_attribute 'rapids-version' '0.15')
+readonly RAPIDS_VERSION=$(get_metadata_attribute 'rapids-version' '0.17')
```
Comment (functicons) [inline rapids/rapids.sh:33]: Why don't use ${DEFAULT_SPARK_RAPIDS_VERSION} as the default?

Comment (medb) [reply rapids/rapids.sh:35]: RAPIDS has a different versioning than Spark RAPIDS

**Comment by medb** [comment]
Installation fails on 2.0 preview images:

```
+ sed -i '/pyarrow .*/d' /opt/conda/miniconda3/conda-meta/pinned
+ /opt/conda/miniconda3/envs/mamba/bin/mamba install -y -c rapidsai -c nvidia -c conda-forge -c defaults cudatoolkit=10.2 rapids=0.17 -p /opt/conda/miniconda3
rapidsai/noarch           
nvidia/linux-64           
nvidia/noarch             
rapidsai/linux-64         
pkgs/r/noarch             
pkgs/r/linux-64           
pkgs/main/noarch          
conda-forge/noarch        
pkgs/main/linux-64        
conda-forge/linux-64      

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.7.8) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['cudatoolkit=10.2', 'rapids=0.17']

Encountered problems while solving.
Problem: package qt-5.12.9-h763d07f_1 requires icu >=67.1,<68.0a0, but none of the providers can be installed
```

**Comment by mengdong** [comment]
@sameerz , could we roll the rapids-dask version back to 0.15? I can investigate that later, try to get spark updated first.

**Comment by medb** [comment]
After roll back it still fails:

```
+ sed -i '/pyarrow .*/d' /opt/conda/miniconda3/conda-meta/pinned
+ /opt/conda/miniconda3/envs/mamba/bin/mamba install -y -c rapidsai -c nvidia -c conda-forge -c defaults cudatoolkit=10.2 rapids=0.15 -p /opt/conda/miniconda3
nvidia/linux-64           
rapidsai/noarch           
nvidia/noarch             
rapidsai/linux-64         
pkgs/r/noarch             
pkgs/main/noarch          
pkgs/r/linux-64           
pkgs/main/linux-64        
conda-forge/noarch        
conda-forge/linux-64      

                  __    __    __    __
                 /  \  /  \  /  \  /  \
                /    \/    \/    \/    \
███████████████/  /██/  /██/  /██/  /████████████████████████
              /  / \   / \   / \   / \  \____
             /  /   \_/   \_/   \_/   \    o \__,
            / _/                       \_____/  `
            |/
        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗
        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗
        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║
        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║
        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║
        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝

        mamba (0.7.8) supported by @QuantStack

        GitHub:  https://github.com/mamba-org/mamba
        Twitter: https://twitter.com/QuantStack

█████████████████████████████████████████████████████████████


Looking for: ['cudatoolkit=10.2', 'rapids=0.15']

Encountered problems while solving.
Problem: package rapids-0.15.0-cuda11.0_py38_g75ae45a_0 requires cuspatial 0.15.*, but none of the providers can be installed
```

**Comment by mengdong** [comment]
My guess is that updated dataproc 2.0 images added new conda dependencies. Let me see what I can do.

**Comment by mengdong** [comment]
@medb since existing dask init action is probably already failing. Could we merge this given all spark test has passed? I will create an issue to track and fix the conda errors.

**Comment by mengdong** [comment]
@sameerz could you change this https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/rapids/rapids.sh#L112 to 0.17, this is the cudf version associated with spark

**Comment by medb** [comment]
Sounds good, I will merge it then.

**Review by medb — APPROVED** [review]

**Comment by bradmiro** [comment]
@mengdong I just confirmed Dask is not broken, but think I have a fix for the RAPIDS issue.

**Comment by mengdong** [comment]
Thank you very much, Brad!

**Pull request merged**